api/agent/agent.go

```diff
@@ -683,21 +685,60 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 			state.UpdateState(ctx, ContainerStateIdle, call.slots)
 			s := call.slots.queueSlot(&hotSlot{done, errC, container, nil})
 
-			select {
-			case <-s.trigger:
-			case <-time.After(time.Duration(call.IdleTimeout) * time.Second):
-				if call.slots.ejectSlot(s) {
-					logger.Info("Canceling inactive hot function")
-					return
-				}
-			case <-ctx.Done(): // container shutdown
-				if call.slots.ejectSlot(s) {
-					return
+			isFrozen := false
```

Comment:
Here can we choose to freeze as soon as container becomes idle for a more strict/robust freezer policy. But at the cost of slowing down a busy slot queue (where dequeue takes < 50 msec.)